### ISSUE TITLE
Document the design difference between `confidence` and `similarity_score`

### ISSUE THREAD

**Issue by rsgoncalves** [mapping-commons/sssom#248]
For example, we have a tool that outputs a mapping "confidence" score, which is a similarity score computed by one of multiple supported similarity metrics (such as edit distance etc).

**Comment by matentzn** [comment]
Ok, let's iterate:

- `confidence`: The degree to which a `mapping_provider` believes that the `mapping_justification` provides evidence that the `mapping` holds true. (Convoluted definition, but it is important to document the confidence pertains to the `mapping_justification`, _not_ the `mapping`).
- `similarity_score`: A measure that, given a specific `similarity_metric`, describes the similarity of the subject and the object.

Note that the `confidence` can be low, while the `similarity_score` can be high (and vice versa), depending on the threshold used. For example, two strings can be similar by 80% Levenshtein distance, but my confidence in that being an indicator of a "correct" mapping is low, say 60%.

The truth is, these two measures are not often distinguished, in particular, if you look at it from the narrow perspective of tool development. Ontology matching tools only provide 1 score - often a "complex" similarity measure that is an aggregate of multiple individual similarity scores. The goal of this score is usually exactly to _approximate_ confidence. 

Now, when combining human and tool work, the difference between the two becomes more apparent.

What do you think?

**Comment by graybeal** [comment]
I think you’ve helped distinguish but left a hole. 

Imagine I’m manually mapping terms. I want to capture how confident I am in the mapping. This is clearly appropriate as a confidence number. But it isn’t a confidence in the justification, in my view—it’s confidence in the actual mapping. 

A computational mapping may provide a similar metric, intended to capture the confidence —calculated algorithmically—that the mapping is correct. 

In both cases, the ‘justification’ may not exist, be named, or even be well understood (think expert judgment or opaque AI algorithms. And it isn’t used to weigh the confidence, the same justification may result in different confidence numbers.

**Comment by matentzn** [comment]
@graybeal I agree that this (confidence in the mapping rather than in the justification) is how we would mostly _interpret_ the confidence score. However, conceptually speaking, I don't think its a good idea to link the confidence directly to the mapping, as it makes it really confusing how to interpret different justifications (some automated some manual) for the same mapping. So I would suggest that conceptually (by that I mean, datamodel wise) we keep associating the confidence with the justification, but interpret the confidence as "matching process (prov:Activity) resulted in a mapping which the mapping process owner (creator, prov:Agent) has X % confidence about".
